hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconNodeManager.java

```diff
@@ -321,4 +321,18 @@ public long getNodeDBKeyCount() throws IOException {
       return nodeCount;
     }
   }
+
+  /**
+   * Remove an existing node from the NodeDB. Explicit removal from admin user.
+   * First this API call removes the node info from NodeManager memory and
+   * if successful, then remove the node finally from NODES table as well.
+   *
+   * @param datanodeDetails Datanode details.
+   * @throws IOException, NodeNotFoundException
+   */
+  public void removeNode(DatanodeDetails datanodeDetails) throws IOException, NodeNotFoundException {
+    getNodeStateManager().removeNode(datanodeDetails);
```

Comment:
IMO, first can remove from Db and then from cache, so that if db update fails, operation can be retried to remove.

Reply:
Ok done.